test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -137,6 +138,13 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, nil
 	}
 
+	// Check if control plane is initialized, otherwise return and wait for the control plane to be available.
```

Comment:
I'm not sure this is the right place to implement this, because this will prevent also the first control-plane machine to be created.

What about moving this test inside the following if https://github.com/kubernetes-sigs/cluster-api/blob/9e2a7c7a886a9e3a06e7b67d9a2e4661991eaf5c/test/infrastructure/docker/controllers/dockermachine_controller.go#L198-L202, given that WaitingForControlPlaneAvailable is a nuance of WaitingForBootstrapDataReason